extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -64,24 +133,79 @@ private String getMongoClientName(Bean bean) {
     @Override
     public HealthCheckResponse call() {
         HealthCheckResponseBuilder builder = HealthCheckResponse.named("MongoDB connection health check").up();
-        Document command = new Document("ping", 1);
-        for (Map.Entry<String, MongoClient> client : clients.entrySet()) {
-            try {
-                Document document = client.getValue().getDatabase("admin").runCommand(command);
-                builder.up().withData(client.getKey(), document.toJson());
-            } catch (Exception e) {
-                return builder.down().withData("reason", "client [" + client.getKey() + "]: " + e.getMessage()).build();
-            }
+        List<Uni<Tuple2<String, String>>> unis = new ArrayList<>();
+        for (Supplier<Uni<Tuple2<String, String>>> client : checks) {
+            unis.add(client.get());
         }
-        for (Map.Entry<String, ReactiveMongoClient> client : reactiveClients.entrySet()) {
-            try {
-                Document document = client.getValue().getDatabase("admin").runCommand(command).await().indefinitely();
-                builder.up().withData(client.getKey(), document.toJson());
-            } catch (Exception e) {
-                return builder.down().withData("reason", "reactive client [" + client.getKey() + "]: " + e.getMessage())
-                        .build();
+
+        if (unis.isEmpty()) {
+            return builder.build();
+        }
+
+        return Uni.combine().all().unis(unis)
+                .collectFailures() // We collect all failures to avoid partial responses.
+                .combinedWith(new Function<List<?>, HealthCheckResponse>() {
+                    @Override
+                    public HealthCheckResponse apply(List<?> list) {
+                        return MongoHealthCheck.this.combine(list, builder);
+                    }
+                }).await().indefinitely(); // All checks fails after a timeout, so if won't be forever.
+
+    }
+
+    @SuppressWarnings("unchecked")
+    private HealthCheckResponse combine(List<?> results, HealthCheckResponseBuilder builder) {
+        for (Object result : results) {
+            Tuple2<String, String> tuple = (Tuple2<String, String>) result;
+            if ("ok".equalsIgnoreCase(tuple.getItem2())) {
+                builder.withData(tuple.getItem1(), "OK");
+            } else {
+                builder.down()
+                        .withData(tuple.getItem1(), "KO, reason: " + tuple.getItem2());
             }
         }
         return builder.build();
     }
+
+    private class MongoHealthConfig implements Supplier<Uni<Tuple2<String, String>>> {
```

Comment:
This is not really a config object but a check right ?
Anyway, naming is hard, we can live with this one ;)

Reply:
you are right! Let me change that!